mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/OpponentList.kt

```diff
@@ -0,0 +1,117 @@
+package ch.epfl.sdp.mobile.ui.prepare_game
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.compositeOver
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.toColor
+import ch.epfl.sdp.mobile.ui.social.Person
+
+/**
+ * Composable that represents a list of potential opponents, with one potentially selected
+ * @param modifier [Modifier] for this composable
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun <P : Person> OpponentList(
+    opponents: List<P>,
+    state: PrepareGameScreenState<P>,
+    modifier: Modifier = Modifier,
+    lazyListState: LazyListState = rememberLazyListState(),
+    key: ((P) -> Any)? = null,
+) {
+  LazyColumn(
+      verticalArrangement = Arrangement.spacedBy(8.dp),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = modifier,
+      contentPadding = PaddingValues(8.dp),
+      state = lazyListState,
+  ) {
+    items(
+        items = opponents,
+        key = key,
+        itemContent = { item ->
+          Opponent(
+              item,
+              onClick = {
+                if (state.selectedOpponent == item) {
+                  state.selectedOpponent = null
+                } else {
+                  state.selectedOpponent = item
+                }
+              },
+              modifier = Modifier.animateItemPlacement(),
+              selected = item == state.selectedOpponent,
+          )
+        },
+    )
+  }
+}
+
+/**
+ * Composable that represents an opponent in [OpponentList]
+ * @receiver The profile of the opponent
+ * @param selected indicates if the opponent is selected
+ * @param onClick button action
+ * @param modifier [Modifier] for this composable
```

Comment:
The fields aren't in the right order, and there's no `@receiver` anymore.

Reply:
Fixed in [3b6d9d6](https://github.com/epfl-SDP/android/pull/200/commits/3b6d9d609f7b69efe8b9d29e5165b1e8cd0b7a60)